Upgrade OPM version used

twasyl opened the pull request:
<!--

Welcome to the Operator SDK! Before contributing, make sure to:

- Read the contributing guidelines https://github.com/operator-framework/operator-sdk/blob/master/CONTRIBUTING.MD
- Rebase your branch on the latest upstream master
- Link any relevant issues, PR's, or documentation
- When fixing an issue, add "Closes #<ISSUE_NUMBER>"
- Follow the below checklist if making a user-facing change 

-->

**Description of the change:**
The generated `Makefile` uses a more recent version of `opm`

**Motivation for the change:**
On OSX, using the version `v1.15.1` of `opm` leads to the following error:
```
Error: Binary was compiled with 'CGO_ENABLED=0', go-sqlite3 requires cgo to work.
```
when doing a `make catalog-build`. Using the latest version of `opm` (`v1.19.1` at the time of this PR) solves the issue.

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [x] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [x] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

twasyl commented:
> @twasyl unfortunately until [operator-framework/operator-registry#710](https://github.com/operator-framework/operator-registry/pull/710) is released this can't merge, since all versions after 1.15.1 do not support multiple linux arches. It will be merged soon though, so we can get this in then. In the meantime, can you add a DCO to your commit message? Thanks!

@estroz Sure, it's not an urgent PR as the version used can be changed in the generated `Makefile`.

camilamacedo86 commented:
@twasyl 

It is only missing add [changelog entry](https://github.com/operator-framework/operator-sdk/blob/master/changelog/fragments/00-template.yaml) to generate the release notes.

twasyl commented:
Hi @camilamacedo86 - Thanks for the review. I've updated the OPM version used to `v1.18.0` and updated `changelog/fragments/00-template.yaml` directly but I'm not sure it's the right way of doing it. Maybe I should rather create a new file somewhere? Please let me know.

joelanford commented:
@twasyl The `00-template.yaml` file is not intended to be changed. Please copy it to a new file in that same folder. Filename doesn't matter as long as it ends with `.yaml`.

twasyl commented:
@joelanford - Thanks for the explanation. I reverted the change to the original file and created a new one for the release notes.

rashmigottipati commented:
https://github.com/operator-framework/operator-registry/pull/710 is merged. Is this PR still blocked?

Also, DCO has to be added to the commit message.

camilamacedo86 commented on internal/plugins/manifests/v2/init.go:
```diff
@@ -32,7 +32,7 @@ import (
 // Version of `opm` to download and use for building index images.
 // This version's release artifacts *must* contain a binary for multiple arches; certain releases do not.
-const opmVersion = "v1.15.1"
+const opmVersion = "v1.18.0"
```
The latest one shows to be 1.19.1 WDYT about updating it?

twasyl replied:
You're right. Done.

varshaprasad96 commented on changelog/fragments/01-upgrade-opm-version.yaml:
```diff
@@ -0,0 +1,8 @@
+entries:
+  - description: >
+      Upgrade OPM version that is used by default in the generated Makefile so `make catalog-build` also works on OSX.
```
Just a nit: would be helpful to specify the version when changelog is generated :)
```suggestion
      Upgrade OPM version that is used by default to x.y.z in the generated Makefile so `make catalog-build` also works on OSX.
```

twasyl replied:
I specified the version there. Thanks

jmrodri reviewed (approved):
/lgtm

camilamacedo86 reviewed (approved):
/lgtm
